clang/test/SemaCXX/cheri/polymorphic-builtins.cpp

```diff
@@ -0,0 +1,124 @@
+/// Check that (most) CHERI builtins are overloaded to allow any capability
```

Comment:
Cap we hoist out a bunch of C-only tests just for sanity, and keep fancy templated/nullptr/etc C++ ones here? I'd be very surprised if we managed to end up with C++ working and not C, but you never know.

Reply:
I'll add C codegen tests, I think __is_same doesn't work in C mode.